src/streaming/controllers/AbrController.js

```diff
@@ -247,22 +247,19 @@ function AbrController() {
         let configRatio = settings.get().streaming.abr.initialRepresentationRatio[type];
 
         if (configBitrate === -1) {
-            const s = { streaming: { abr: { initialBitrate: {}}}};
             if (configRatio > -1) {
                 const representation = adapter.getAdaptationForType(0, type).Representation;
                 if (Array.isArray(representation)) {
                     const repIdx = Math.max(Math.round(representation.length * configRatio) - 1, 0);
-                    s.streaming.abr.initialBitrate[type] = representation[repIdx].bandwidth;
+                    configBitrate = representation[repIdx].bandwidth;
                 } else {
-                    s.streaming.abr.initialBitrate[type] = 0;
+                    configBitrate = 0;
                 }
             } else if (!isNaN(savedBitrate)) {
-                s.streaming.abr.initialBitrate[type] = savedBitrate;
+                configBitrate = savedBitrate;
             } else {
-                s.streaming.abr.initialBitrate[type] = (type === Constants.VIDEO) ? DEFAULT_VIDEO_BITRATE : DEFAULT_AUDIO_BITRATE;
+                configBitrate = (type === Constants.VIDEO) ? DEFAULT_VIDEO_BITRATE : DEFAULT_AUDIO_BITRATE;
             }
-            settings.update(s);
```

Comment:
By removing those lines, datas will not be saved between different streaming.....no?

Reply:
The problem I see with using settings to persist an initialBitrate that is not really the one set by the user but a "calculated one" (ex: coming from DOMStorage) is just that. When playing a second stream, even when local storage is disabled, AbrController will set as initial bitrate the one that was set for the first played stream which for most of cases is the value coming from DOMStorage.

If user set initialBitrate through settings, with this approach it will be saved across consecutive playbacks.